bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+: [TC] originally called `scriptSig`. The Bitcoin code base used Hungarian notation when it was published.  This transaction field appearing in Transaction Inputs was named for containing the signature and being of the type script.  Today, this field may be better understood as the _input script_ which is used to satisfy spending conditions for non-segwit outputs, redirects validation to the Witness Stack for wrapped segwit outputs, and is empty for native segwit outputs. The Input Script may contain an actual script when spending a Bare Output Script, but may only contain push operations for any output type since P2SH.
+
+; Leaf Script
+: [TC] A Condition Script used for spending P2TR outputs via the script path.
+
+; Leaf Version
+: [TC] The version of Script used in a leaf script. At time of writing, only _Tapscript_ is defined.
+
+; Locktime
+: [TC] Part of the Transaction Header. The locktime specifies a time (in blockheight or UNIX time) before which a transaction cannot be included in a block if any input's sequence number indicates non-finality.  The locktime can also indicate relative spending lock in combination with `OP_CHECKSEQUENCEVERIFY`. Serialized as the four byte <code>nLockTime</code> integer field.
+
+; Marker
+: [SA] 1-byte serialization artifact indicating that a type of extended serialization is being used for this transaction.  Must always be <code>00</code> (see BIP144).  (Note: Since the serialization of a non-segwit transaction would have the input counter at the marker's position, a whole segwit transaction will always appear invalid to non-segwit nodes since the input counter must be greater than zero.)
+
+; Outpoint
+: [TC] Identifies the UTXO being spent in a transaction input.  Consists of a txid and Output Index.
+
+; Output Counter
+: [SA] The length of the transaction output list.
+
+; Output Index
+: [TC] Part of an Outpoint. The position of the output in a transaction's Output List that created the identified UTXO.
+
+; Output Script
+: [TC] originally called `scriptPubKey`. The part of a transaction output that will contain either a Condition Script or a Forwarding Script. Originally, the Bitcoin code base used Hungarian notation. This field was presumably named `scriptPubKey` to refer to the transaction field of the type _script_ that contained the public key. The use of this field had evolved beyond that interpretation even when Bitcoin was published, since the field could already contain more complicated scripts.  Today, this field is referred to as the _Output Script_.
+
+; P2SH Program
+: [AT] A script template that forwards input validation to the Redeem Script.  P2SH Programs are a type of Forwarding Script.
+
+; Redeem Script
+: [AT] The Condition Script used for spending P2SH outputs. Note that the Input Script of wrapped segwit inputs is not a Redeem Script but a Witness Program.
+
+; Script Arguments
+: [AT] Arguments presented to satisfy the Condition Script, including Signatures, and Public Keys
```

Comment:
```suggestion
: [AT] Arguments presented to satisfy the Condition Script, including Signatures, and Public Keys.
```
(A few other definitions below are missing the period.)

Reply:
I’ve deliberately left them out when it’s a single line and only used them for multi sentence descriptions.

Reply:
In hindsight that was not an accurate statement, but I have since improved the offending lines in this regard ^^